### PR TITLE
iceoryx: fix cross-build to iOS/tvOS/watchOS + several fixes

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

Additional fixes and features:
- add missing official CMake imported targets
- dependencies declaration in `requirements()`, not `config_options()` !
- add `fPIC` option
- don't remove root package folder from builddirs (potential side effetcs)
- properly propagate libcxx (except for Windows obviously, it was broken for all OS different than Linux, Macos and Neutrino).

Formatting:
- restructure recipe to avoid unecessary indirections, and have methods/properties close to their usage.

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
I did not change validate(), but several verifications are suspicious:
- I hardly doubt that limitations regarding gcc are limites to Linux
- why libc++ is not supported here with clang?

**Comment by dotChris90** [comment]
> @dotChris90
> 
> I did not change `validate()`, but several verifications are suspicious:
> 
> * I hardly doubt that limitations regarding gcc are specific to Linux
> * why libc++ is not supported here with clang?

ok checked again in container for clang10,clang9 - you are right there is no longer a reason for that. the reason before was because i didnt see the reason behind the error "just working for libstdc++11" (but i see you already patched it) and now working fine with libc++

**Comment by dotChris90** [comment]
interesting - as far as i can see the linking error for clang7 just happens if shared=True. and just during linking executable routi. maybe missed again libatomic dependency or sth in this way. somehow clang doesnt "autolink" as many libs as gcc do.

**Comment by dotChris90** [comment]
btw the conan jenkins servers also targeting iOS/tvOS/watchOS? how amazing simple.

**Comment by SpaceIm** [comment]
> btw the conan jenkins servers also targeting iOS/tvOS/watchOS? how amazing simple.

No yet, but I try to fix as many recipes as I can for iOS.

**Review by SSE4 — APPROVED** [review]

**Review by czoido — APPROVED** [review]

File: recipes/iceoryx/all/conanfile.py [hunk 1]
```diff
@@ -13,113 +16,79 @@ class IceoryxConan(ConanFile):
...
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        # Honor fPIC option
+        for cmake_file in [
+            os.path.join("iceoryx_binding_c", "CMakeLists.txt"),
+            os.path.join("iceoryx_posh", "CMakeLists.txt"),
+            os.path.join("iceoryx_utils", "CMakeLists.txt"),
+        ]:
+            tools.replace_in_file(os.path.join(self._source_subfolder, cmake_file),
```
Comment (uilianries) [inline recipes/iceoryx/all/conanfile.py:90]: Nice!

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**